### ISSUE TITLE
Handle `x-rate-limit-type: service`

### ISSUE THREAD

**Issue by MingweiSamuel** [MingweiSamuel/Riven#30]
https://discord.com/channels/187652476080488449/187652476080488449/875851922299568139

**Comment by MingweiSamuel** [comment]
Formatted example response from above:
```js
RiotApiError {
    reqwest_error: reqwest::Error {
        kind: Status(429),
        url: Url {
            scheme: "https",
            cannot_be_a_base: false,
            username: "",
            password: None,
            host: Some(Domain("euw1.api.riotgames.com")),
            port: None,
            path: "/lol/league/v4/entries/by-summoner/xEblhubzDNCyCdjyTNF6fIPotR3Mkt21eCSnEeET2PLOX7A",
            query: None,
            fragment: None
        }
    },
    retries: 3,
    response: Some(Response {
        url: Url {
            scheme: "https",
            cannot_be_a_base: false,
            username: "",
            password: None,
            host: Some(Domain("euw1.api.riotgames.com")),
            port: None,
            path: "/lol/league/v4/entries/by-summoner/xEblhubzDNCyCdjyTNF6fIPotR3Mkt21eCSnEeET2PLOX7A",
            query: None,
            fragment: None
        },
        status: 429,
        headers: {
            "content-type": "application/json;charset=utf-8",
            "date": "Fri, 13 Aug 2021 21:18:46 GMT",
            "retry-after": "3",
            "x-app-rate-limit": "3000:10,180000:600",
            "x-app-rate-limit-count": "18:10,58:600",
            "x-method-rate-limit": "300:60",
            "x-method-rate-limit-count": "13:60",
            "x-rate-limit-type": "service",
            "x-riot-edge-trace-id": "52326b42-b25a-4e28-8ad0-c38d692459e6",
            "content-length": "62",
            "connection": "keep-alive"
        }
    }),
    status_code: Some(429)
}
```

**Comment by MingweiSamuel** [comment]
Fixed by 5c26ddd2115716890cdca123da25fb66738b3de0
